solang-parser/src/doccomment.rs

```diff
@@ -0,0 +1,172 @@
+use crate::pt::Comment;
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum DocComment {
+    Line { comment: DocCommentTag },
+    Block { comments: Vec<DocCommentTag> },
+}
+
+impl DocComment {
+    pub fn comments(&self) -> Vec<&DocCommentTag> {
+        match self {
+            DocComment::Line { comment } => vec![comment],
+            DocComment::Block { comments } => comments.iter().collect(),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct DocCommentTag {
+    pub tag: String,
+    pub tag_offset: usize,
+    pub value: String,
+    pub value_offset: usize,
+}
+
+enum CommentType {
+    Line,
+    Block,
+}
+
+/// For the start to end offset, filter all the doc comemnts comments and parse them into
```

Comment:
Shouldn't this be `From the start to end`?

Reply:
Thanks for catching that, fixed.